File: src/code-samples/deepagents/frontend-todo-list-setup.ts
// :snippet-start: frontend-todo-list-setup-js
import { createDeepAgent } from "deepagents";
import { todoListMiddleware } from "langchain";

const agent = await createDeepAgent({
  model: "anthropic:claude-sonnet-4-6",
  middleware: [todoListMiddleware()],
});
// :snippet-end:

// :remove-start:
if (!agent) {
  throw new Error("agent not created");
}
console.log("✓ frontend-todo-list-setup sample validated");
// :remove-end:

File: src/code-samples/deepagents/frontend-todo-list-setup.py
"""Frontend todo list: enable TodoListMiddleware on create_deep_agent."""

# :snippet-start: frontend-todo-list-setup-py
from deepagents import create_deep_agent
from langchain.agents.middleware import TodoListMiddleware

agent = create_deep_agent(
    model="anthropic:claude-sonnet-4-6",
    middleware=[TodoListMiddleware()],
)
# :snippet-end:

# :remove-start:
assert agent is not None
print("✓ frontend-todo-list-setup sample validated")
# :remove-end:
